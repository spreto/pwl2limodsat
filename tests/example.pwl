c
c Hand generated.
c spreto 05/11/2020
c Piecewise Linear Function pwl file example
c

pwl

c
c Boundary Prototypes
c

b 8 -9 -6
b 1 -2 1
b 0.33333 0 -1
b 0 1 0
b 1 -1 0
b 0 0 1
b 1 0 -1

c
c First linear piece with boundaries
c

p 4 9 0 1 2 3
g 1
g 3
g 4
g 6

c
c Second linear piece with boundaries
c

p 5 6 0 1 -1 2
g 2
l 3
g 4
g 7

c
c Third linear piece with boundaries
c

p 4 3 -1 1 0 1
l 1
l 2
g 5
g 6
