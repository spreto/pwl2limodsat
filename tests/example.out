-= Formula phi =-

Unit 1 :: Clause      :: 19 
Unit 2 :: Clause      :: 10 12 
Unit 3 :: Clause      :: 21 
Unit 4 :: Implication :: 2 3
Unit 5 :: Negation    :: 4
Unit 6 :: Minimum     :: 1 5
Unit 7 :: Clause      :: 28 
Unit 8 :: Minimum     :: 6 7
Unit 9 :: Clause      :: 10 12 
Unit 10 :: Clause      :: 21 
Unit 11 :: Implication :: 9 10
Unit 12 :: Negation    :: 11
Unit 13 :: Clause      :: 19 
Unit 14 :: Minimum     :: 12 13
Unit 15 :: Clause      :: 28 
Unit 16 :: Minimum     :: 14 15
Unit 17 :: Maximum     :: 8 16
Unit 18 :: Clause      :: 28 
Unit 19 :: Clause      :: 19 
Unit 20 :: Minimum     :: 18 19
Unit 21 :: Clause      :: 10 12 
Unit 22 :: Clause      :: 21 
Unit 23 :: Implication :: 21 22
Unit 24 :: Negation    :: 23
Unit 25 :: Minimum     :: 20 24
Unit 26 :: Maximum     :: 17 25

-= MODSAT Set Phi =-

-= Linear Piece 1 =-
Formula 1:
Unit 1 :: Clause      :: 4 
Unit 2 :: Clause      :: 3 
Unit 3 :: Equivalence :: 1 2
Formula 2:
Unit 1 :: Clause      :: 5 
Unit 2 :: Clause      :: 4 4 
Unit 3 :: Equivalence :: 1 2
Formula 3:
Unit 1 :: Clause      :: 6 
Unit 2 :: Clause      :: 5 5 
Unit 3 :: Equivalence :: 1 2
Formula 4:
Unit 1 :: Clause      :: 7 
Unit 2 :: Clause      :: 6 6 
Unit 3 :: Equivalence :: 1 2
Formula 5:
Unit 1 :: Clause      :: 8 
Unit 2 :: Clause      :: 7 7 
Unit 3 :: Equivalence :: 1 2
Formula 6:
Unit 1 :: Clause      :: 3 
Unit 2 :: Clause      :: 4 8 
Unit 3 :: Negation    :: 2
Unit 4 :: Equivalence :: 1 3
Formula 7:
Unit 1 :: Clause      :: 10 
Unit 2 :: Clause      :: 9 
Unit 3 :: Equivalence :: 1 2
Formula 8:
Unit 1 :: Clause      :: 11 
Unit 2 :: Clause      :: 10 10 
Unit 3 :: Equivalence :: 1 2
Formula 9:
Unit 1 :: Clause      :: 12 
Unit 2 :: Clause      :: 11 11 
Unit 3 :: Equivalence :: 1 2
Formula 10:
Unit 1 :: Clause      :: 9 
Unit 2 :: Clause      :: 10 12 
Unit 3 :: Negation    :: 2
Unit 4 :: Equivalence :: 1 3
Formula 11:
Unit 1 :: Clause      :: 14 
Unit 2 :: Clause      :: 13 
Unit 3 :: Equivalence :: 1 2
Formula 12:
Unit 1 :: Clause      :: 15 
Unit 2 :: Clause      :: 14 14 
Unit 3 :: Equivalence :: 1 2
Formula 13:
Unit 1 :: Clause      :: 16 
Unit 2 :: Clause      :: 15 15 
Unit 3 :: Equivalence :: 1 2
Formula 14:
Unit 1 :: Clause      :: 15 16 
Unit 2 :: Clause      :: 2 
Unit 3 :: Equivalence :: 1 2
Formula 15:
Unit 1 :: Clause      :: 13 
Unit 2 :: Clause      :: 9 
Unit 3 :: Implication :: 1 2
Formula 16:
Unit 1 :: Clause      :: 18 
Unit 2 :: Clause      :: 6 15 
Unit 3 :: Clause      :: -17 17 
Unit 4 :: Negation    :: 3
Unit 5 :: Implication :: 2 4
Unit 6 :: Negation    :: 5
Unit 7 :: Equivalence :: 1 6
Formula 17:
Unit 1 :: Clause      :: 19 
Unit 2 :: Clause      :: 18 18 
Unit 3 :: Equivalence :: 1 2

-= Linear Piece 2 =-
Formula 1:
Unit 1 :: Clause      :: 20 
Unit 2 :: Clause      :: 20 
Unit 3 :: Negation    :: 2
Unit 4 :: Equivalence :: 1 3
Formula 2:
Unit 1 :: Clause      :: 22 
Unit 2 :: Clause      :: 21 
Unit 3 :: Equivalence :: 1 2
Formula 3:
Unit 1 :: Clause      :: 23 
Unit 2 :: Clause      :: 22 22 
Unit 3 :: Equivalence :: 1 2
Formula 4:
Unit 1 :: Clause      :: 23 
Unit 2 :: Clause      :: 2 
Unit 3 :: Equivalence :: 1 2
Formula 5:
Unit 1 :: Clause      :: 21 
Unit 2 :: Clause      :: 20 
Unit 3 :: Implication :: 1 2

-= Linear Piece 3 =-
Formula 1:
Unit 1 :: Clause      :: 25 
Unit 2 :: Clause      :: 24 
Unit 3 :: Equivalence :: 1 2
Formula 2:
Unit 1 :: Clause      :: 26 
Unit 2 :: Clause      :: 25 25 
Unit 3 :: Equivalence :: 1 2
Formula 3:
Unit 1 :: Clause      :: 26 
Unit 2 :: Clause      :: 1 
Unit 3 :: Equivalence :: 1 2
Formula 4:
Unit 1 :: Clause      :: 24 
Unit 2 :: Clause      :: 20 
Unit 3 :: Implication :: 1 2
Formula 5:
Unit 1 :: Clause      :: 27 
Unit 2 :: Clause      :: 12 
Unit 3 :: Clause      :: 24 
Unit 4 :: Implication :: 2 3
Unit 5 :: Negation    :: 4
Unit 6 :: Equivalence :: 1 5
Formula 6:
Unit 1 :: Clause      :: 28 
Unit 2 :: Clause      :: 27 27 
Unit 3 :: Equivalence :: 1 2
